Packet++/src/DnsResourceData.cpp

```diff
@@ -74,7 +74,7 @@ bool IPv4DnsResourceData::toByteArr(uint8_t* arr, size_t &arrLength, IDnsResourc
 
 	uint32_t addrAsInt = m_Data.toInt();
 	arrLength = sizeof(addrAsInt);
-	memcpy(arr, &addrAsInt, arrLength);
+	memcpy(arr, &addrAsInt, sizeof addrAsInt);
```

Comment:
why does this change matter?

Reply:
The `arrLength` is a reference to external variable so its value have to read before calling the `memcpy` while the expression `sizeof addrAsInt` is a constant so the call of `memcpy` can be optimized down to the only instruction `mov 4,(place-in-memory)` instead of `call memcpy`. The function that can be inlined is being optimized more effective when we pass the constant value instead of a variable in arguments.

Reply:
The arrLength is a reference to external variable so its value have to read before calling the memcpy while the expression sizeof addrAsInt is a constant. All the same as for Packet's constructor.

Reply:
I think this is over-optimization... beside, `arrLength` is anyway set in the line before

Reply:
Yes you're right. But if we append later some new code (I think it's highly unlikely) between these lines the compiler can generate not optimal binary code. Another benefit of this change is the source code is more readable because it is not necessary to find the place where `arrLength` is assigned.

Reply:
I'll leave this decision to you. But if you choose to leave it as is, please change `sizeof addrAsInt` to `sizeof(addrAsInt)` to comply with other places in the code where `sizeof` is used